src/General/Button/Button.test.tsx

```diff
@@ -15,6 +15,17 @@ it('<Button> should render with text "click me" and an onClick handler', () => {
   expect(ButtonSnapshot).toMatchSnapshot();
 });
 
+it('<Button> should render as ther given tag', () => {
```

Comment:
This test works fine but just some minor points:
- You don't have to use `data-test-id` here. You only use it when you want to use `getByTestId`
- Instead of `.not.toBeEmpty()` you could use `toBeTruthy` https://jestjs.io/docs/en/expect#tobetruthy

That said I think this test should 1. test more tags and 2. be more explicit that we are checking for the tag name. I suggest something like this instead.
```
describe('contains the correct className for each variant:', () => {
  const tags = ['a', 'h1', 'div'];
  tags.forEach((tag: string & Element) => {
    it(`<Button> should render as the given tag: ${tag}`, () => {
      const { container } = render(
        <Button tag={tag as React.ElementType}>anchor button</Button>
      );
      const tagRegexp = new RegExp(tag, 'i');
      expect(container.querySelector(tag).tagName).toMatch(tagRegexp);
    });
  });
});
```

Also, I think we should add one test for the default case where no tag is passed and it should render a `button` element.

PS: Not sure if `tag: string & Element` is bad typescript haha. You could just change it to `any` as well and remove the `React.ElementType`.

Reply:
> Not sure if tag: string & Element
I just changed it to `string` since that's what is being iterated.

> add one test for the default case
Done!

> be more explicit that we are checking for the tag name
Done!